src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -285,7 +300,133 @@ public Response replaceFileInDataset(
         }
             
     } // end: replaceFileInDataset
+    
+    @POST
+    @Path("{id}/metadata")
+    public Response updateFileMetadata(@FormDataParam("jsonData") String jsonData,
+                    @PathParam("id") String fileIdOrPersistentId
+        ) throws DataFileTagException, CommandException { //MAD: should catch these?
+        
+        try {
+            DataverseRequest req = createDataverseRequest(findUserOrDie());
+            final DataFile df = execCommand(new GetDataFileCommand(req, findDataFileOrDie(fileIdOrPersistentId)));
+
+            //Much of this code is taken from the replace command, simplified as we aren't actually switching files
+            User authUser;
+            try {
+                authUser = findUserOrDie();
```

Comment:
Kind of a nit here: if you're only using authUser fro findUserOrDie there's no need to declare it outside the try.

Reply:
True!